include/libopencm3/stm32/common/dac_common_all.h

```diff
@@ -322,105 +132,114 @@ Unmask bits [(n-1)..0] of LFSR/Triangle Amplitude equal to (2**(n+1)-1
 
 /* --- DAC_DHR8RD values --------------------------------------------------- */
-#define DAC_DHR8RD_DACC2DHR_LSB		(1 << 8)
-#define DAC_DHR8RD_DACC2DHR_MSK		(0x00FF << 8)
-#define DAC_DHR8RD_DACC1DHR_LSB		(1 << 0)
-#define DAC_DHR8RD_DACC1DHR_MSK		(0x00FF << 0)
+#define DAC_DHR8RD_DACC2DHR_SHIFT	8
+#define DAC_DHR8RD_DACC2DHR_MSK		0xFF
+#define DAC_DHR8RD_DACC1DHR_SHIFT	0
+#define DAC_DHR8RD_DACC1DHR_MSK		0xFF
 
 
 /* --- DAC_DOR1 values ----------------------------------------------------- */
-#define DAC_DOR1_DACC1DOR_LSB		(1 << 0)
-#define DAC_DOR1_DACC1DOR_MSK		(0x0FFF << 0)
+#define DAC_DOR1_DACC1DOR_SHIFT		0
+#define DAC_DOR1_DACC1DOR_MSK		0xFFF
 
 
 /* --- DAC_DOR2 values ----------------------------------------------------- */
-#define DAC_DOR2_DACC2DOR_LSB		(1 << 0)
-#define DAC_DOR2_DACC2DOR_MSK		(0x0FFF << 0)
+#define DAC_DOR2_DACC2DOR_SHIFT		0
+#define DAC_DOR2_DACC2DOR_MSK		0xFFF
+
+/* --- DAC_SR values ------------------------------------------------------- */
 
-/** @defgroup dac_sr_values DAC_SR Values
-@{*/
 /** DAC channel 1 DMA underrun flag */
-#define DAC_SR_DMAUDR1		(1 << 13)
+#define DAC_SR_DMAUDR1			(1 << 13)
 
 /** DAC channel 2 DMA underrun flag */
-#define DAC_SR_DMAUDR2		(1 << 29)
-/**@}*/
+#define DAC_SR_DMAUDR2			(1 << 29)
+
+/* --- Function prototypes ------------------------------------------------- */
 
 /** DAC channel identifier */
-typedef enum {
-	CHANNEL_1, CHANNEL_2, CHANNEL_D
-} data_channel;
+#define DAC_CHANNEL1		(1 << 0)
+#define DAC_CHANNEL2		(1 << 1)
+#define DAC_CHANNEL_BOTH	(DAC_CHANNEL1 | DAC_CHANNEL2)
 
 /** DAC data size (8/12 bits), alignment (right/left) */
-typedef enum {
-	RIGHT8, RIGHT12, LEFT12
-} data_align;
-
-/* --- Function prototypes ------------------------------------------------- */
+enum dac_align {
+	DAC_ALIGN_RIGHT8,
+	DAC_ALIGN_RIGHT12,
+	DAC_ALIGN_LEFT12,
+};
+
+/* DAC waveform generation options. */
+enum dac_wave {
+	DAC_WAVE_DISABLE  = 0,
+	DAC_WAVE_NOISE    = 1,
+	DAC_WAVE_TRIANGLE = 2,
+	DAC_WAVE_SAWTOOTH = 3,
+};
 
 BEGIN_DECLS
 
-void dac_enable(data_channel dac_channel);
-void dac_disable(data_channel dac_channel);
-void dac_buffer_enable(data_channel dac_channel);
-void dac_buffer_disable(data_channel dac_channel);
-void dac_dma_enable(data_channel dac_channel);
-void dac_dma_disable(data_channel dac_channel);
-void dac_trigger_enable(data_channel dac_channel);
-void dac_trigger_disable(data_channel dac_channel);
-void dac_set_trigger_source(uint32_t dac_trig_src);
-void dac_set_waveform_generation(uint32_t dac_wave_ens);
-void dac_disable_waveform_generation(data_channel dac_channel);
-void dac_set_waveform_characteristics(uint32_t dac_mamp);
-void dac_load_data_buffer_single(uint16_t dac_data, data_align dac_data_format,
-				 data_channel dac_channel);
-void dac_load_data_buffer_dual(uint16_t dac_data1, uint16_t dac_data2,
-			       data_align dac_data_format);
-void dac_software_trigger(data_channel dac_channel);
+void dac_enable(uint32_t dac, uint8_t channel);
```

Comment:
hrmm, you _probably_ just want "int" for the channel, there's no need for it to be explicitly 8bit, which can require more work

Reply:
Ok, will fix.